File: Level 2/영어 끝말잇기.js
/*
 * 문제 출처: 프로그래머스 코딩 테스트 연습, https://programmers.co.kr/learn/challenges
 */

function solution(n, words) {
    let answer = [0,0];
    let last = words[0][words[0].length-1];
    let hs = new Set();
    hs.add(words[0]);
    for (let i = 1; i < words.length; i++) {
        if (words[i][0] != last || hs.has(words[i])) {
            answer[0] = i%n+1;
            answer[1] = parseInt(i/n)+1;
            break;
        }
        last = words[i][words[i].length-1];
        hs.add(words[i]);
    }
    return answer;
}

File: Level 2/영어 끝말잇기.java
/**
 * 문제 출처: 프로그래머스 코딩 테스트 연습, https://programmers.co.kr/learn/challenges
 */
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = {0, 0};
        char last = words[0].charAt(words[0].length()-1);
        HashSet<String> hs = new HashSet<>();
        hs.add(words[0]);
        for (int i = 1; i < words.length; i++) {
            if (words[i].charAt(0) != last || hs.contains(words[i])) {
                answer[0] = i%n+1;
                answer[1] = i/n+1;
                break;
            }
            last = words[i].charAt(words[i].length()-1);
            hs.add(words[i]);
        }

        return answer;
    }
}
